### PR TITLE
add `addIndex` & `dropIndex` @ `AlterTableBuilder`

### PR DESCRIPTION
**Pull request by Gaspero**
closes https://github.com/kysely-org/kysely/issues/507
closes https://github.com/kysely-org/kysely/issues/508

- [x]  implement
- [x]  apply suggestions from previous PR https://github.com/kysely-org/kysely/pull/522
- [x]  test
- [x]  document

### CODE REVIEW

File: src/schema/alter-table-builder.ts [hunk 1]
```diff
@@ -239,6 +242,24 @@ export class AlterTableBuilder implements ColumnAlteringInterface {
     })
   }
 
+  addIndex(indexName: string): AlterTableAddIndexBuilder {
```
Comment (igalklebanov) [inline src/schema/alter-table-builder.ts:245]: missing jsdocs here. you know the usual description + example.

File: src/schema/alter-table-builder.ts [hunk 2]
```diff
@@ -239,6 +242,24 @@ export class AlterTableBuilder implements ColumnAlteringInterface {
...
+    return new AlterTableAddIndexBuilder({
+      ...this.#props,
+      node: AlterTableNode.cloneWithTableProps(this.#props.node, {
+        addIndex: AddIndexNode.create(indexName),
+      }),
+    })
+  }
+
+  dropIndex(indexName: string): AlterTableExecutor {
```
Comment (igalklebanov) [inline src/schema/alter-table-builder.ts:254]: missing jsdocs here. you know the usual description + example.

**Review by igalklebanov — CHANGES_REQUESTED** [review]
Hey 👋 

Looks good! 💪 

Missing some jsdocs..

File: src/schema/alter-table-builder.ts (hunk 1 above)
Comment (Gaspero) [reply src/schema/alter-table-builder.ts:264]: Done ✅ https://github.com/kysely-org/kysely/pull/720/commits/ea88a775c627880032ac46b7352954ff54306cd3

File: src/schema/alter-table-builder.ts (hunk 2 above)
Comment (Gaspero) [reply src/schema/alter-table-builder.ts:290]: Done ✅ https://github.com/kysely-org/kysely/pull/720/commits/ea88a775c627880032ac46b7352954ff54306cd3

**Pull request merged**